probatus/missing/imputation.py

```diff
@@ -0,0 +1,318 @@
+                reproducible results set it to integer.
+        """
+        self.clf = clf
+        self.model_na_support = model_na_support
+        self.scorer = get_single_scorer(scoring)
+        self.strategies = strategies
+        if cv is None:
+            self.cv = 5
+        else :
+            self.cv = cv
+        self.verbose = verbose
+
+        self.n_jobs = n_jobs
+        
+        if random_state is None:
+            self.random_state = 42
+        else:
+            self.random_state = random_state
+
+        self.fitted = False
+        self.report = pd.DataFrame([])
+
+    def __repr__(self):
+        return "Imputation comparision for {}".format(self.clf.__class__.__name__)
+
+
+    def fit(self, X, y,column_names=None,categorical_columns='auto'):
+        """
+        Calculates the cross validated results for various imputation strategies.
+        
+        Args:
+            X (pd.DataFrame):
+                input variables.
+
+            y (pd.Series):
+                target variable.
+
+            column_names (None, or list of str, optional):
+                List of feature names for the dataset. 
+                If None, then column names from the X dataframe are used.
+            
+            categorical_features (None, or list of str, optional):default=auto
+                List of categorical features.The imputation strategy for categorical 
+                is different that compared to numerical features. If auto try to infer
+                the categorical columns based on 'object' and 'category' datatypes.
+        """
+        #Place holder for results.
+        results = []
+
+        self.X, self.column_names = preprocess_data(X, column_names=column_names,
+                                                          verbose=self.verbose)
+        self.y = preprocess_labels(y, index=self.X.index, verbose=self.verbose)
+                                                         
+
+        #Identify categorical features if not explicitly specified.
+        if 'auto' in categorical_columns:
+            X_cat = X.select_dtypes(include=['category','object'])
+            categorical_columns = X_cat.columns.to_list()
+            for column in categorical_columns:
+                X[column] = X[column].astype('category')
+        else :
+            #Check if the passed columns are in the dataframe.
+            assert categorical_columns in X.columns,"All categorical columns not in the dataframe."
+            X_cat = X[categorical_columns]
+        #Identify the numeric columns.Numeric columns are all columns expect the categorical
+        # columns
+        X_num = X.drop(columns = categorical_columns,inplace=False)
+        numeric_columns = X_num.columns.to_list()
+        
+        for strategy in self.strategies:
+
+            numeric_transformer = Pipeline(steps=[
+                    ('imputer', self.strategies[strategy])])
+
+            categorical_transformer = Pipeline(steps=[
```

Comment:
What happens if user provides as clf the pipeline that handles categorical features? Do we see it as an option that we want to support?

Reply:
As of now, the missing values categorical features are handled internally with Onehotencoding. As of now we can go with the default approach and make it more complex if there is demand for it.